### DISCUSSION TITLE
Proper settings for immediate missing writer notification.

### DISCUSSION THREAD

**Discussion by xumepoc** [OpenDDS/OpenDDS#4264] [Q&A] [answered]
I have several domains and one is responsible for nodes management. Each node has its writer and reader. In order to keep track of which node is online or offline I have implemented a BIT listener which uses the USER_DATA as additional information about the node ID. So when a node's writer comes online I receive on_data_available and if I get DDS::RETCODE_OK the node is online and I read the USER_DATA. If the node shuts down improperly (crash) I receive  on_data_available with DDS::NOT_ALIVE_DISPOSED_INSTANCE_STATE. But the problem is that this is not immediate. The BIT listeners receives the on_data_available 5 minutes later. The QoS for the topic, writer and reader for all nodes is:
```
qos.liveliness.lease_duration.sec = 1;
qos.liveliness.lease_duration.nanosec = 0;
```  

I also have implemented 
```
conn_retry_initial_delay=100
conn_retry_backoff_multiplier=0.5
conn_retry_attempts=2
passive_reconnect_duration=400
```

But nothing is changed. What is the proper setting in order for the BIT listener to receive immediately the on_data_available with DDS::NOT_ALIVE_DISPOSED_INSTANCE_STATE?

**Comment by jrw972** [comment, marked as answer]
The liveliness settings for individual writers and readers have no impact on the BITs.

Given that you say you receiving `on_data_available` 5 minutes after a node goes offline, that leads me to believe you are using RtpsDiscovery because the default participant lease duration is 5 minutes.  Each participant sends SPDP messages periodically to indicate that it is still alive.  It also declares a lease duration which tells its peers that they should declare it dead if they don't see a message for this amount of time.  If you are in fact using RtpsDiscovery, you can look at the [`ResendPeriod` and `LeaseDuration` configuration values](https://opendds.readthedocs.io/en/latest-release/devguide/run_time_configuration.html#configuring-for-ddsi-rtps-discovery) and tune them appropriately.

**Comment by xumepoc** [comment]
Yes, sorry I complitly forgot to specify the Discovery method. I am indeed using Rtps. 
LeaseDuration did the trick, thank you so much :)
